File: MantenimientoDaxDirectorio/MantenimientoDirectorio/EscojeCamposDir.vb
Option Strict Off
Option Explicit On
Imports System.Windows.Forms
Imports System.Data.SqlClient
Imports DattCom
Friend Class EscojeCamposDir
    Dim CamposAntes As String = ""
    Dim LimSup As Integer = 0
    Dim LimInf As Integer = 0
    Dim MatCamposAntes() As String
    Dim CamposCol As String = ""
    Dim Sistema As String = ""

    Public Function EscojeCampos(ByRef Sistem As String) As String
        '    If CamposCol = Nothing Then Return ""
        Sistema = Sistem
        Me.ShowDialog()
        SaveSetting(Sistema, "Dir", "Campos", CamposCol)
        EscojeCampos = CamposCol
    End Function

    Private Sub EscojeCamposDir_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Load
        'Dim prog As New DaxLib.DaxLibMalla
        'prog.ColorF(Me)
        ' prog = Nothing

        CamposAntes = GetSetting(Sistema, "Dir", "Campos", "")
        If CamposAntes > "" Then
            MatCamposAntes = Split(CamposAntes, ",")
            LimSup = UBound(MatCamposAntes)
            LimInf = LBound(MatCamposAntes)
        End If
        llenarMalla()
    End Sub
    Private Sub llenarMalla()
        Dim sSQL As String = "select top 1 * from directorioList "
        Dim conn As New SqlClient.SqlConnection(datosEmpresa.strConxAdcom)
        Dim I As Integer
        conn.Open()
        Dim misqlDa As New SqlDataAdapter(sSql, conn)
        Dim dato As New DataTable("datos")
        misqlDa.Fill(dato)
        With malla
            For I = 0 To dato.Columns.Count - 1
                .Items.Add(dato.Columns(i).ColumnName())
            Next
        End With

        'Dim rs As New DataTable
        'Dim DR As New SqlClient.SqlDataReader
        'Dim CONN As 
        'Dim prog As New DaxData.DaxLibDatos
        ''Dim ConxAdcom As New ADODB.Connection
        ''Dim PROG As New DaxLibBases
        'Dim i As Integer
        'Dim j As Integer

        ''ConxAdcom.ConnectionString = Prog.StrAdcom    'prog.ArmStr(Emp.NombreBaseAlex, Emp.servidor, "", Emp.ClaveBD, Emp.UsuarioBd)
        ''ConxAdcom.Open

        ''rs = prog.DaxData("MSSQL", "sp_columns directorioadc") ' ConxAdcom.Execute("sp_columns directorioadc")
        'With malla
        '    Do Until rs.EOF
        '        If Not IsDBNull(rs.Fields("column_name").Value) Then .Items.Add(rs.Fields("column_name").Value)
        '        rs.MoveNext()
        '    Loop
        'End With
        If LimSup > 0 Then
            For i = 0 To malla.Items.Count - 1
                For j = 1 To LimSup
                    If malla.Items(i) = MatCamposAntes(j) Then malla.SetItemChecked(i, True)
                Next j
            Next i
        End If
        malla.SetItemChecked(0, True)
    End Sub


    Private Sub Command2_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles Command2.Click
        Dim i As Integer
        CamposCol = "Codigo"
        With malla
            For i = 1 To .Items.Count - 1
                If malla.GetItemChecked(i) = True Then
                    If CamposCol > "" Then CamposCol = CamposCol & ","
                    CamposCol = CamposCol & .Items.Item(i).ToString
                End If
            Next i
        End With
        Me.Close()
    End Sub

    Private Sub Command1_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles Command1.Click
        CamposCol = ""
        SaveSetting(Sistema, "Dir", "Campos", CamposCol)
        Me.Close()
    End Sub

    Private Sub Command3_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Command3.Click
        Me.Close()
    End Sub
End Class

File: MantenimientoDaxDirectorio/MantenimientoDirectorio/EscojeCamposDir.cs
using System;
using System.Data;
using System.Data.SqlClient;
using DattCom;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;

namespace directMnt
{
	internal partial class EscojeCamposDir
	{
		private string CamposAntes = "";
		private int LimSup = 0;
		private int LimInf = 0;
		private string[] MatCamposAntes;
		private string CamposCol = "";
		private string Sistema = "";

		public EscojeCamposDir()
		{
			InitializeComponent();
		}

		public string EscojeCampos(ref string Sistem)
		{
			string EscojeCamposRet = default;
			// If CamposCol = Nothing Then Return ""
			Sistema = Sistem;
			ShowDialog();
			Interaction.SaveSetting(Sistema, "Dir", "Campos", CamposCol);
			EscojeCamposRet = CamposCol;
			return EscojeCamposRet;
		}

		private void EscojeCamposDir_Load(object sender, EventArgs e)
		{
			// Dim prog As New DaxLib.DaxLibMalla
			// prog.ColorF(Me)
			// prog = Nothing

			CamposAntes = Interaction.GetSetting(Sistema, "Dir", "Campos", "");
			if (Operators.CompareString(CamposAntes, "", false) > 0)
			{
				MatCamposAntes = Strings.Split(CamposAntes, ",");
				LimSup = Information.UBound(MatCamposAntes);
				LimInf = Information.LBound(MatCamposAntes);
			}
			llenarMalla();
		}
		private void llenarMalla()
		{
			string sSQL = "select top 1 * from directorioList ";
			var conn = new SqlConnection(datosEmpresa.strConxAdcom);
			int I;
			conn.Open();
			var misqlDa = new SqlDataAdapter(sSQL, conn);
			var dato = new DataTable("datos");
			misqlDa.Fill(dato);
			{
				var withBlock = malla;
				var loopTo = dato.Columns.Count - 1;
				for (I = 0; I <= loopTo; I++)
					withBlock.Items.Add(dato.Columns[I].ColumnName);
			}

			// Dim rs As New DataTable
			// Dim DR As New SqlClient.SqlDataReader
			// Dim CONN As 
			// Dim prog As New DaxData.DaxLibDatos
			// 'Dim ConxAdcom As New ADODB.Connection
			// 'Dim PROG As New DaxLibBases
			// Dim i As Integer
			// Dim j As Integer

			// 'ConxAdcom.ConnectionString = Prog.StrAdcom    'prog.ArmStr(Emp.NombreBaseAlex, Emp.servidor, "", Emp.ClaveBD, Emp.UsuarioBd)
			// 'ConxAdcom.Open

			// 'rs = prog.DaxData("MSSQL", "sp_columns directorioadc") ' ConxAdcom.Execute("sp_columns directorioadc")
			// With malla
			// Do Until rs.EOF
			// If Not IsDBNull(rs.Fields("column_name").Value) Then .Items.Add(rs.Fields("column_name").Value)
			// rs.MoveNext()
			// Loop
			// End With
			if (LimSup > 0)
			{
				var loopTo1 = malla.Items.Count - 1;
				for (I = 0; I <= loopTo1; I++)
				{
					for (int j = 1, loopTo2 = LimSup; j <= loopTo2; j++)
					{
						if (Conversions.ToBoolean(Operators.ConditionalCompareObjectEqual(malla.Items[I], MatCamposAntes[j], false)))
							malla.SetItemChecked(I, true);
					}
				}
			}
			malla.SetItemChecked(0, true);
		}


		private void Command2_Click(object sender, EventArgs e)
		{
			int i;
			CamposCol = "Codigo";
			{
				var withBlock = malla;
				var loopTo = withBlock.Items.Count - 1;
				for (i = 1; i <= loopTo; i++)
				{
					if (malla.GetItemChecked(i) == true)
					{
						if (Operators.CompareString(CamposCol, "", false) > 0)
							CamposCol = CamposCol + ",";
						CamposCol = CamposCol + withBlock.Items[i].ToString();
					}
				}
			}
			Close();
		}

		private void Command1_Click(object sender, EventArgs e)
		{
			CamposCol = "";
			Interaction.SaveSetting(Sistema, "Dir", "Campos", CamposCol);
			Close();
		}

		private void Command3_Click(object sender, EventArgs e)
		{
			Close();
		}
	}
}